qctrlopencontrols/dynamic_decoupling_sequences/predefined.py

```diff
@@ -59,21 +70,63 @@ def _add_pre_post_rotations(
         resulting after the addition of pi/2 pulses at the start and end of the sequence.
     """
 
+    # These settings correspond to the case where the pulses of the sequence
+    # already yield an identity, so that we want the the pi/2-pulses to cancel
+    # each other out
+    rabi_value = np.pi / 2
+    initial_azimuthal = 0
+    final_azimuthal = np.pi
+    detuning_value = 0
+
+    # Count the number of X, Y, and Z pi-pulses
+    x_pi_pulses = np.count_nonzero(np.logical_and(np.isclose(rabi_rotations, np.pi),
+                                                  np.isclose(azimuthal_angles, 0.)))
+    y_pi_pulses = np.count_nonzero(np.logical_and(np.isclose(rabi_rotations, np.pi),
+                                                  np.isclose(azimuthal_angles, np.pi/2.)))
+    z_pi_pulses = np.count_nonzero(np.logical_and(np.isclose(rabi_rotations, 0.),
+                                                  np.isclose(detuning_rotations, np.pi)))
+
```

Comment:
That's a very good idea. Instead of using `assert` I used the exception that was usually thrown in that module, so if someone passes an unacceptable sequence the kind of error will be more consistent with the what they might see elsewhere.

I also made the checks for X/Y/Z pulses more strict to make sure they aren't being misidentified